org.mixedrealitytoolkit.uxcore/Button/PressableButton.cs

```diff
@@ -623,6 +681,85 @@ private float IntersectQuad(Vector3 start, Vector3 end, float depth)
             }
         }
 
+        /// <summary>
+        /// Registers the duple Collider + XRBaseInteractor as triggering proximity.
+        /// </summary>
+        /// <param name="collider">Collider triggering proximity.</param>
+        /// <param name="xrBaseInteractor">Interactor triggering proximity.</param>
+        public void OnProximityEntered(ProximityEnteredEventArgs proximityEnteredEventArgs)
+        {
+            if (proximityEnteredEventArgs.collider != null &&
+                activeCollidersWithInteractor.Add((proximityEnteredEventArgs.collider, proximityEnteredEventArgs.interactor)) &&
+                activeCollidersWithInteractor.Count >= 1)
+            {
+                isInProximity = true;
+                ProximityHoverEntered?.Invoke(this, proximityEnteredEventArgs, null);
+            }
+        }
+
+        /// <summary>
+        /// Unregisters the duple Collider + XRBaseInteractor as triggering proximity.
+        /// </summary>
+        /// <param name="collider">Collider that in combination with the interactor was triggering proximity.</param>
+        /// <param name="xrBaseInteractor">Interactor that in combination with the collider was triggering proximity.</param>
+        public void OnProximityExited(ProximityExitedEventArgs proximityExitedEventArgs)
+        {
+            if (proximityExitedEventArgs.collider != null &&
+                activeCollidersWithInteractor.Remove((proximityExitedEventArgs.collider, proximityExitedEventArgs.interactor)) &&
+                activeCollidersWithInteractor.Count == 0)
+            {
+                isInProximity = false;
+                ProximityHoverExited?.Invoke(this, proximityExitedEventArgs, null);
+            }
+        }
+
+        /// <summary>
+        /// Method invoked when a Proximity or a Hovered entered events are triggered.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="proximityEnteredEventArgs">ProximityEnteredEventArgs if the event was triggered by proximity, null otherwise.</param>
+        /// <param name="hoverEnterEventArgs">HoverEnterEventArgs if the event was triggered by hovering, null otherwise.</param>
+        private void OnProximityHoverEntered(object sender, ProximityEnteredEventArgs proximityEnteredEventArgs, HoverEnterEventArgs hoverEnterEventArgs)
```

Comment:
I don't think `OnProximityHoverEntered` `OnProximityHoverExited` and `SetEnabledDynamicComponents` are now required since we decided to use events only.

Reply:
Removed when removed ProximityHoveringDynamicComponents array.